// node-hill-s/dist/src/util/keys/whitelisted.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KeyTypes = void 0;
/**
 * An enum that contains a list of all the keys allowed in player.keypress.
 *
 * This is meant to verify packets to ensure that only valid key types are being sent.
 */
var KeyTypes;
(function (KeyTypes) {
    KeyTypes["alphabetical"] = "a-z";
    KeyTypes["numerical"] = "0-9";
    KeyTypes["shift"] = "shift";
    KeyTypes["lshift"] = "lshift";
    KeyTypes["rshift"] = "rshift";
    KeyTypes["control"] = "control";
    KeyTypes["lcontrol"] = "lcontrol";
    KeyTypes["rcontrol"] = "rcontrol";
    KeyTypes["alt"] = "alt";
    KeyTypes["ralt"] = "ralt";
    KeyTypes["lalt"] = "lalt";
    KeyTypes["tab"] = "tab";
    KeyTypes["space"] = "space";
    KeyTypes["return"] = "return";
    KeyTypes["enter"] = "enter";
    KeyTypes["backspace"] = "backspace";
    KeyTypes["mouse1"] = "mouse1";
    KeyTypes["mouse2"] = "mouse2";
    KeyTypes["mouse3"] = "mouse3";
    KeyTypes["up"] = "up";
    KeyTypes["down"] = "down";
    KeyTypes["left"] = "left";
    KeyTypes["right"] = "right";
})(KeyTypes = exports.KeyTypes || (exports.KeyTypes = {}));
const keys = "abcdefghijklmnopqrstuvwxyz0123456789".split("");
keys.push("shift", "lshift", "rshift", "control", "lcontrol", "rcontrol", "alt", "ralt", "lalt", "tab", "space", "return", "enter", "backspace", "mouse1", "mouse2", "mouse3", "up", "down", "left", "right");
exports.default = keys;
